File: ProtocolTestManager/PTMService/Common/Types/Adapter.cs
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

namespace Microsoft.Protocols.TestManager.PTMService.Common.Types
{
    public class Adapter
    {
        public string Name { get; set; }

        public string DisplayName { get; set; }

        // Used by managed adapter only, the type name of adapter implementation.
        public string AdapterType { get; set; }

        public AdapterKind Kind { get; set; }

        public AdapterKind[] SupportedKinds { get; set; }

        public string ScriptDirectory { get; set; }

        public string ShellScriptDirectory { get; set; }
    }
}

File: ProtocolTestManager/PTMService/PTMService/ClientApp/src/model/Adapter.ts
// Copyright (c) Microsoft. All rights reserved.
// Licensed under the MIT license. See LICENSE file in the project root for full license information.

export interface Adapter {
  // Adapter name
  Name: string
  // Adapter displayname
  DisplayName: string
  // Customize Adapter implementation type, only
  AdapterType: string
  // Supported Kinds
  SupportedKinds: AdapterKind[]
  // Adapter Kind
  Kind: AdapterKind
  // PowerShell adapter Script Directory
  ScriptDirectory: string
  // Shell adapter Script Directory
  ShellScriptDirectory: string
}

export enum AdapterKind {
  Managed = 'Managed',
  PowerShell = 'PowerShell',
  Shell = 'Shell',
  Interactive = 'Interactive',
}

export enum ChangedField {
  AdapterKind,
  AdapterType,
  ScriptDirectory,
  ShellScriptDirectory
}

export interface AdapterChangedEvent {
  Field: ChangedField
  NewValue: string | number | undefined
  Adapter: string
}
